src/main/java/io/gravitee/common/templating/FreeMarkerComponent.java

```diff
@@ -28,143 +28,103 @@
 import java.util.Locale;
 import java.util.Map;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * @author Antoine CORDIER (antoine.cordier at graviteesource.com)
  * @author GraviteeSource Team
  *
  * Utility component that compiles freemarker templates from various source to various outputs.
  *
- * This can be used in plugins as it handles template loading from the parent classloader.
- *
- * To be resolved from the class path, templates *must* reside in the `freemarker` resources package.
+ * If looking for templates in the classPath, it is the responsibility of the caller to provide a classLoader.
  */
+@Slf4j
 public class FreeMarkerComponent {
 
-    private static final Logger LOG = LoggerFactory.getLogger(FreeMarkerComponent.class);
-
-    private static final String TEMPLATES_BASE = "/freemarker";
-
     private final Configuration freeMarkerConfiguration;
 
     /**
-     * Create a FreeMarker component with default configuration, meaning that templates are loaded from
-     * a freemarker resources package located in the classpath, using the current classloader and its parent.
-     * The parent classloader takes precedence over the current classloader.
-     * {@see #freemarker.cache.ClassTemplateLoader(Class, String)}
-     */
-    public FreeMarkerComponent() {
-        this("");
-    }
-
-    /**
-     * Convenience constructor to create a FreeMarker component which will load templates from a subpackage
-     * inside the freemarker resources package.
-     * {@link FreeMarkerComponent#FreeMarkerComponent()}
-     */
-    public FreeMarkerComponent(String subPackagePath) {
-        var basePackagePath = sanitizeBasePackage(subPackagePath);
-        this.freeMarkerConfiguration = freemarkerConfiguration(basePackagePath);
-    }
-
-    /**
-     * Creates a FreeMarker component which will look for templates from a given path on the file system first,
-     * and fallback to the classpath loader.
-     */
-    public FreeMarkerComponent(Path path) {
-        this(path, "");
-    }
-
-    /**
-     * Creates a FreeMarker component which will look for templates from a given path on the file system first,
-     * and fallback to the freemarker resources subpackage.
-     * {@link FreeMarkerComponent#FreeMarkerComponent(String)}
-     * {@link FreeMarkerComponent#FreeMarkerComponent(Path)}
+     * Creates a new FreeMarkerComponent.
+     * @param path path to a template directory on the File System
+     * @param classLoader class loader to load templates from the class path
+     * @param classTemplateLoaderBase base path to load templates from the class path
+     *
+     * If path is defined (not null), templates will be looked for on the file system first, then in the class path if
+     * a class loader has been provided.
+     *
+     * If classTemplateLoaderBase is not defined (null), templates will be looked for in the root of the class path.
```

Comment:
Does it mean that if I use the builder like this:

`FreeMarkerComponent.builder().classLoader(classLoader).build()`, then I can evaluate a template with `freemarkerComponent.generateFromTemplate("freemarker/test.tpl", ...)` --> note doesn't start with '/'

But if I use it like this `FreeMarkerComponent.builder().classLoader(classLoader).classTemplateLoaderBase("/freemarker").build()` then I use it like this `freemarkerComponent.generateFromTemplate("/test.tpl", ...)` --> note the '/' at the beginning

If so, do you think we can try to align the usage by always expecting something starting with '/'?

Reply:
FreeMarker handles that for us. e.g.

I can use either one of
  - `classTemplateLoaderBase("freemarker/")`
  - `classTemplateLoaderBase("freemarker")`
  - `classTemplateLoaderBase("freemarker/")`
  
And get the same result with:
  - `freemarkerComponent.generateFromTemplate("/test.tpl", data)`
  
Which leaves us with a consistent behaviour